soc/silabs/silabs_siwx917/siwg917/nwp_init.c

```diff
@@ -0,0 +1,96 @@
+	static Z_DECL_ALIGN(struct _isr_list) Z_GENERIC_SECTION(.intList)                          \
+		__used __isr_##name = {irq, flags, expression, arg}
+
+#if IS_ENABLED(CONFIG_WIFI_SIWX917) && IS_ENABLED(CONFIG_BT_SIWX917)
+#define NWP_INIT_COEX_MODE SL_SI91X_WLAN_BLE_MODE
+#elif IS_ENABLED(CONFIG_WIFI_SIWX917)
+#define NWP_INIT_COEX_MODE SL_SI91X_WLAN_ONLY_MODE
+#elif IS_ENABLED(CONFIG_BT_SIWX917)
+#define NWP_INIT_COEX_MODE SL_SI91X_BLE_MODE
+#else
+#error "Not supported configuration!"
+#endif
+
+static int silabs_siwx917_nwp_init(void)
+{
+	sl_wifi_device_configuration_t network_config = {
+		.boot_option = LOAD_NWP_FW,
+		.mac_address = NULL,
+		.band = SL_SI91X_WIFI_BAND_2_4GHZ,
+		.region_code = DEFAULT_REGION,
+		.boot_config = {
+			.oper_mode = SL_SI91X_CLIENT_MODE,
+			.coex_mode = NWP_INIT_COEX_MODE,
+			.feature_bit_map = SL_SI91X_FEAT_SECURITY_OPEN | SL_SI91X_FEAT_WPS_DISABLE,
+			.tcp_ip_feature_bit_map = SL_SI91X_TCP_IP_FEAT_EXTENSION_VALID,
+			.ext_tcp_ip_feature_bit_map = SL_SI91X_CONFIG_FEAT_EXTENSION_VALID,
+			.config_feature_bit_map = SL_SI91X_ENABLE_ENHANCED_MAX_PSP,
+			.custom_feature_bit_map = SL_SI91X_CUSTOM_FEAT_EXTENSION_VALID,
+			.ext_custom_feature_bit_map =
+				MEMORY_CONFIG | SL_SI91X_EXT_FEAT_XTAL_CLK |
+				SL_SI91X_EXT_FEAT_FRONT_END_SWITCH_PINS_ULP_GPIO_4_5_0,
+		}};
+	sl_si91x_boot_configuration_t *cfg = &network_config.boot_config;
+
+#if IS_ENABLED(CONFIG_WIFI_SIWX917)
+	cfg->tcp_ip_feature_bit_map |=
+#if IS_ENABLED(CONFIG_NET_IPV6)
+		SL_SI91X_TCP_IP_FEAT_DHCPV6_CLIENT | SL_SI91X_TCP_IP_FEAT_IPV6 |
+#endif
+		SL_SI91X_TCP_IP_FEAT_DHCPV4_CLIENT | SL_SI91X_TCP_IP_FEAT_DNS_CLIENT |
+		SL_SI91X_TCP_IP_FEAT_SSL | SL_SI91X_TCP_IP_FEAT_MDNSD | SL_SI91X_TCP_IP_FEAT_ICMP;
+	cfg->ext_tcp_ip_feature_bit_map |=
+		SL_SI91X_EXT_TCP_IP_WINDOW_SCALING | SL_SI91X_EXT_TCP_IP_TOTAL_SELECTS(10);
+#endif
+
+#if IS_ENABLED(CONFIG_BT_SIWX917)
+	cfg->ext_custom_feature_bit_map |= SL_SI91X_EXT_FEAT_BT_CUSTOM_FEAT_ENABLE;
+	cfg->bt_feature_bit_map |= SL_SI91X_BT_RF_TYPE | SL_SI91X_ENABLE_BLE_PROTOCOL;
+	cfg->ble_feature_bit_map |= SL_SI91X_BLE_MAX_NBR_PERIPHERALS(RSI_BLE_MAX_NBR_PERIPHERALS) |
+				    SL_SI91X_BLE_MAX_NBR_CENTRALS(RSI_BLE_MAX_NBR_CENTRALS) |
+				    SL_SI91X_BLE_MAX_NBR_ATT_SERV(RSI_BLE_MAX_NBR_ATT_SERV) |
+				    SL_SI91X_BLE_MAX_NBR_ATT_REC(RSI_BLE_MAX_NBR_ATT_REC) |
+				    SL_SI91X_BLE_PWR_INX(RSI_BLE_PWR_INX) |
+				    SL_SI91X_BLE_PWR_SAVE_OPTIONS(RSI_BLE_PWR_SAVE_OPTIONS) |
+				    SL_SI91X_916_BLE_COMPATIBLE_FEAT_ENABLE |
+				    SL_SI91X_FEAT_BLE_CUSTOM_FEAT_EXTENSION_VALID;
+	cfg->ble_ext_feature_bit_map |= SL_SI91X_BLE_NUM_CONN_EVENTS(RSI_BLE_NUM_CONN_EVENTS) |
+					SL_SI91X_BLE_NUM_REC_BYTES(RSI_BLE_NUM_REC_BYTES) |
+					SL_SI91X_BLE_ENABLE_ADV_EXTN |
+					SL_SI91X_BLE_AE_MAX_ADV_SETS(RSI_BLE_AE_MAX_ADV_SETS);
+#endif
+
+	return sl_wifi_init(&network_config, NULL, sl_wifi_default_event_handler);
```

Comment:
If I understand right there is dependency between the calls to `sl_wifi_init()` and `sl_net_set_profile()`. Documentation says:

> sl_net_init() should be called before [sl_net_set_profile()]

However, `sl_net_init()` call `sl_net_set_profile()` and then `sl_wifi_init()`.

Maybe it would more reasonable to replace this call to `sl_wifi_init()` by `sl_net_init()`  and avoid possible trouble in the future?

Reply:
The sl_net component is not really needed with BT so I don't prefer using it.
Furthermore we don't use (directly or indirectly) the sl_net_get/delete_profile APIs so I believe the it would be better just to remove sl_net_set_profile() from the WiFi.
What do you think?

Reply:
I agree. We just have to check it still works. I suggest to add a `FIXME` above the call to `sl_net_set_profile()` and fix it later.